crates/wascc-provider/src/lib.rs

```diff
@@ -25,10 +26,20 @@ type EnvVars = std::collections::HashMap<String, String>;
 /// TODO: In the future, we will look at loading capabilities using the "sidecar" metaphor
 /// from Kubernetes.
 #[derive(Clone)]
-pub struct WasccProvider {}
+pub struct WasccProvider<S> {
+    store: S,
+}
+
+impl<S: ModuleStore + Send + Sync> WasccProvider<S> {
+    /// Returns a new wasCC provider configured to use the proper data directory
```

Comment:
Note: This is not valid here now, but we will need it anyway when we implement logging